receiver/hostmetricsreceiver/internal/scraper/filesystemscraper/filesystem_scraper.go

```diff
@@ -85,8 +99,10 @@ func (s *scraper) Scrape(_ context.Context) (pdata.Metrics, error) {
 
 	if len(usages) > 0 {
 		metrics.EnsureCapacity(metricsLen)
-		initializeFileSystemUsageMetric(metrics.AppendEmpty(), now, usages)
-		appendSystemSpecificMetrics(metrics, now, usages)
+		s.recordFileSystemUsageMetric(now, usages)
+		s.mb.Emit(metrics)
+		s.recordSystemSpecificMetrics(now, usages)
+		s.mb.Emit(metrics)
```

Comment:
s.mb.Emit should be called only once per scrape

Reply:
Then we will change order or metrics, it is acceptable?

Reply:
Yes. Order of metrics is the only change that supposed to happen after switching to the new metrics builder

Reply:
I added a small helper function for tests `findMetricByName()`, if it is not the right way, let me know and I will think more on a solution.

Reply:
Sounds good, thanks